ingress/elb/elb.go

```diff
@@ -131,8 +157,22 @@ func (e *elb) findFrontEndElbs() ([]string, error) {
 	return clusterFrontEnds, nil
 }
 
-func (e *elb) Detach(frontend types.FrontendInput) error {
-	return nil
+// Detach removes this instance from all the front end ELBs
+func (e *elb) Detach() error {
+	var result error
+	for _, elb := range e.elbs {
+		log.Infof("Deregistering instance %s with elb %s", e.instanceID, elb)
+		_, err := e.awsElb.DeregisterInstancesFromLoadBalancer(&aws_elb.DeregisterInstancesFromLoadBalancerInput{
+			Instances:        []*aws_elb.Instance{&aws_elb.Instance{InstanceId: aws.String(e.instanceID)}},
+			LoadBalancerName: aws.String(elb),
+		})
+
+		if err != nil {
+			result = fmt.Errorf("unable to deregister instance %s with elb %s: %v", e.instanceID, elb, err)
```

Comment:
We should probably concat the errors rather than losing them. Like store a list of them and return that. It will be misleading if there are multiple errors.